### PR TITLE
[NET-4103] ci: build s390x

### PR DESCRIPTION
**Pull request by loshz**
### Description

This PR adds build and unit test support for the s390x architecture in Consul Enterprise automated workflows.

The following changes have been made:
- Build the Consul binary for s390x.
- Run tests using a dev build of the above binary.

Tested in ENT: https://github.com/hashicorp/consul-enterprise/pull/6265

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -319,12 +371,13 @@ jobs:
...
     strategy:
       matrix:
-        arch: ["386", "amd64", "arm", "arm64"]
+        arch: ["386", "amd64", "arm", "arm64", "s390x"]
       fail-fast: true
     env:
       version: ${{ needs.set-product-version.outputs.product-version }}
       zip_name: consul_${{ needs.set-product-version.outputs.product-version }}_linux_${{ matrix.arch }}.zip
 
+    if: ${{ endsWith(github.repository, '-enterprise') }} || ${{ matrix.arch != 's390x' }}
```
Comment (jmurret) [inline .github/workflows/build.yml:380]: so we are no longer verifying at all on OSS?  Is this intentional?

Comment (loshz) [reply .github/workflows/build.yml:380]: Yes, the overhead of running this on OSS is not worth the cost due to the fact the customer requiring this specific build is running ENT only.

Comment (jmurret) [reply .github/workflows/build.yml:381]: right sorry, i mis read the `if` and was thinking that no platforms would be verified on OSS.  👍

**Review by jmurret — APPROVED** [review]
LGTM.  Nice work!  :shipit:

**Review by emilymianeil — APPROVED** [review]
🥳 LGTM - just remove that `loshz/NET-4103-build-s390x` from `build.yml` before merging 😄

**Pull request merged**